lib/engine/game/g_18_eu/game.rb

```diff
@@ -991,6 +991,13 @@ def home_token_locations(corporation)
 
           all_minor_hexes(corporation)
         end
+
+        def exchange_corporations(exchange_ability)
+          return super unless exchange_ability.owner.minor?
+
+          parts = graph.connected_nodes(exchange_ability.owner).keys
```

Comment:
this is expensive, can we put a return if game.loading? 

if not, then we need to turn this into an auto action

Reply:
Indeed we can. I forgot that's a thing!

Reply:
I directed it to super because that will evaluate all corps being legal, which I feel is more correct during load.